src/engine/strat_engine/backstore/backstore.rs

```diff
@@ -215,96 +456,373 @@ impl Backstore {
+                            self.cache
+                                .as_mut()
+                                .expect("cache_tier.is_some() <=> self.cache.is_some()");
+                        cache_tier.add(dm, &mut cache_device, paths, force)
+                    }
+                    None => {
+                        // FIXME: This is obviously a bad idea, but it means
+                        // that is necessary to add a pool UUID parameter to
+                        // this method. That course of action would require
+                        // various changes in the calling code which have not
+                        // been agreed upon. See GitHub PR:
+                        // https://github.com/stratis-storage/stratisd/pull/784
+                        let pool_uuid = self.data_tier.block_mgr.pool_uuid();
+                        let bdm =
+                            BlockDevMgr::initialize(pool_uuid, paths, MIN_MDA_SECTORS, force)?;
+
+                        let linear = self.linear
+                            .take()
+                            .expect("cache_tier.is_none() <=> self.linear.is_some()");
+                        let (cache_tier, cache) = CacheTier::new(dm, bdm, linear)?;
+                        self.cache = Some(cache);
+
+                        let uuids = cache_tier
+                            .block_mgr
+                            .blockdevs()
+                            .iter()
+                            .map(|&(uuid, _)| uuid)
+                            .collect::<Vec<_>>();
+
+                        self.cache_tier = Some(cache_tier);
+
+                        Ok(uuids)
+                    }
+                }
+            }
+            BlockDevTier::Data => {
+                self.data_tier
+                    .add(dm, self.cache.as_mut(), self.linear.as_mut(), paths, force)
+            }
+        }
     }
 
-    /// Allocate space from the underlying device.
+    /// Allocate requested chunks from device.
+    /// Returns None if it is not possible to satisfy the request.
+    /// Each segment allocated is contiguous with its neighbors in the return
+    /// vector.
+    /// WARNING: All this must change when it becomes possible to return
+    /// sectors to the store.
     /// WARNING: metadata changing event
     pub fn alloc_space(&mut self, sizes: &[Sectors]) -> Option<Vec<Vec<(Sectors, Sectors)>>> {
-        self.data_tier.alloc_space(sizes)
+        if self.available() < sizes.iter().cloned().sum() {
+            return None;
+        }
+
+        let mut chunks = Vec::new();
+        for size in sizes {
+            chunks.push(vec![(self.next, *size)]);
+            self.next += *size;
+        }
+        Some(chunks)
     }
 
-    /// Return a reference to the blockdevs that form the basis of the
-    /// backstore.
+    /// Return a reference to all the devs that this pool has ownership of.
+    /// This includes blockdevs, cachedevs, any device about which information
```

Comment:
suggest rephrase, this makes it sound like more than just blockdevs and cachedevs are being returned.

Reply:
I'ld rather leave it more open, until we close it off. We may end up having other sorts of devices that we are willing to expose.